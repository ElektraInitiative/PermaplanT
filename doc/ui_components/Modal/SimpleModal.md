# SimpleModal

## Description

A simple text based modal.

## Props

| Name           | Type              | Optional | Description                                                                                                               |
| -------------- | ----------------- | -------- | ------------------------------------------------------------------------------------------------------------------------- |
| title          | string            | no       | Text that will be displayed in the heading of the modal.                                                                  |
| message        | string            | no       | Text that acts as the main component of the modal.                                                                        |
| setShow        | (boolean) => void | no       | Callback that informs the parent when the modal should be hidden/displayed (e.g. when the user pressed the close button). |
| show           | boolean           | no       | Decides whether the modal should be shown.                                                                                |
| cancelBtnTitle | string            | yes      | Label text for the cancel button. The cancel button will only be displayed if this prop is set.                           |
| submitBtnTitle | string            | no       | Label text for the submit button.                                                                                         |
| onCancel       | () => void        | yes      | Click callback for cancel button. Can only be called if cancelBtnTitle is set.                                            |
| onSubmit       | () => void        | no       | Click callback for submit button.                                                                                         |
