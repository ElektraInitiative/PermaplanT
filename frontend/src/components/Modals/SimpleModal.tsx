import SimpleButton from '../Button/SimpleButton';
import TransparentBackground from '../TransparentBackground';
import ModalContainer from './ModalContainer';

interface SimpleModalProps {
  /** Modal headline displayed on top.*/
  title: string;
  /** The text of the modal.*/
  body: string;
  /** Callback that informs the parent when the modal should be hidden/displayed (e.g. when the user pressed the close button).*/
  setShow: (show: boolean) => void;
  /** Decides whether the modal should be shown.*/
  show: boolean;
  /** Click callback for cancel/close button.*/
  onCancel?: () => void;
  /** label of the cancel/abort button. */
  cancelBtnTitle?: string;
  /** label of the submit/ok button. */
  submitBtnTitle: string;
  /** submit button onClick event handler. */
  onSubmit: () => void;
}

/** Simple Modal used for prompts */
export default function SimpleModal({
  title,
  body,
  show,
  setShow,
  cancelBtnTitle,
  submitBtnTitle,
  onCancel,
  onSubmit,
}: SimpleModalProps) {
  return (
    <>
      <TransparentBackground
        onClick={() => {
          setShow(false);
        }}
        show={show}
      />
      <ModalContainer show={show}>
        <div className="flex min-h-[200px] w-[400px] flex-col justify-between rounded-lg bg-neutral-100 p-6 dark:bg-neutral-100-dark">
          <h2>{title}</h2>
          <p className="mb">{body}</p>
          <div className="space-between flex flex-row justify-center space-x-8">
            {cancelBtnTitle && (
              <SimpleButton onClick={onCancel} className="max-w-[240px] grow">
                {cancelBtnTitle}
              </SimpleButton>
            )}
            <SimpleButton onClick={onSubmit} className="max-w-[240px] grow">
              {submitBtnTitle}
            </SimpleButton>
          </div>
        </div>
      </ModalContainer>
    </>
  );
}
