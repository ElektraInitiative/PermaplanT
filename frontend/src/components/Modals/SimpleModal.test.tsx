import SimpleModal from './SimpleModal';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const callback = jest.fn();
  const tree = renderer
    .create(
      <MemoryRouter>
        <SimpleModal
          title="Test"
          body="Test"
          setShow={callback}
          show={true}
          submitBtnTitle={'Ok'}
          cancelBtnTitle={'Cancel'}
          onSubmit={callback}
        />
      </MemoryRouter>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
