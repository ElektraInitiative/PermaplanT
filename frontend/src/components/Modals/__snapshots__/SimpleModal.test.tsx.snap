// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
[
  <div
    className="fixed bottom-0 right-0 z-[1010] "
    onClick={[Function]}
  >
    <div
      style={
        {
          "opacity": 0,
        }
      }
    >
      <div
        className="h-screen w-screen bg-[#000]"
      />
    </div>
  </div>,
  <div
    className="fixed left-[50%] top-[15%] z-[1020]"
    style={
      {
        "opacity": 0,
        "transform": "translateX(-50%) translateZ(0)",
      }
    }
  >
    <div
      className="flex min-h-[200px] w-[400px] flex-col justify-between rounded-lg bg-neutral-100 p-6 dark:bg-neutral-100-dark"
    >
      <h2>
        Test
      </h2>
      <p
        className="mb"
      >
        Test
      </p>
      <div
        className="space-between flex flex-row justify-center space-x-8"
      >
        <button
          className="inline-flex h-10 w-full items-center justify-center rounded-lg text-white px-4 py-2.5 text-center text-sm font-medium focus:outline-none focus:ring-4 focus:ring-blue-300 border bg-primary-500 dark:bg-primary-300 border-primary-500 hover:bg-primary-600 dark:hover:bg-primary-200 active:bg-primary-900 text-primary-50 dark:text-primary-700 max-w-[240px] grow"
          type="button"
        >
          Cancel
        </button>
        <button
          className="inline-flex h-10 w-full items-center justify-center rounded-lg text-white px-4 py-2.5 text-center text-sm font-medium focus:outline-none focus:ring-4 focus:ring-blue-300 border bg-primary-500 dark:bg-primary-300 border-primary-500 hover:bg-primary-600 dark:hover:bg-primary-200 active:bg-primary-900 text-primary-50 dark:text-primary-700 max-w-[240px] grow"
          onClick={[MockFunction]}
          type="button"
        >
          Ok
        </button>
      </div>
    </div>
  </div>,
]
`;
